File: clients/swift/src/Models/Problem.swift
import Foundation

/// RFC 7807 problem detail returned by the API for error responses.
public struct Problem: Codable, Sendable {
    public let type: String
    public let title: String
    public let status: Int
    public let detail: String

    public init(type: String, title: String, status: Int, detail: String) {
        self.type = type
        self.title = title
        self.status = status
        self.detail = detail
    }
}

File: clients/csharp/src/Models/Problem.cs
using System.Text.Json.Serialization;

namespace MyMind.Models;

/// <summary>RFC 7807 problem detail returned by the API for error responses.</summary>
public class Problem
{
    [JsonPropertyName("type")]
    public required string Type { get; init; }

    [JsonPropertyName("title")]
    public required string Title { get; init; }

    [JsonPropertyName("status")]
    public required int Status { get; init; }

    [JsonPropertyName("detail")]
    public required string Detail { get; init; }
}
